### PR TITLE
[Utopia 720] fix search text and filter to and relation instead of or 

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

- fix search text and filter relation when do query

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 1]
```diff
@@ -30,27 +34,65 @@ const PIAList = () => {
...
   const updateSearchUrl = () => {
-    navigate(`?searchText=${searchText}`);
+    setSearchParamsForSearchText();
   };
 
   useEffect(() => {
     const listener = (event: any) => {
       if (event.code === 'Enter' || event.code === 'NumpadEnter') {
         event.preventDefault();
-        navigate(`?searchText=${searchText}`);
```
Comment (binggaofw) [inline src/frontend/src/pages/PIAListPage/index.tsx:42]: I did not find a good way to call a function outside of useEffect hook. that's why duplicate code here. please let me know if you know how to do it

File: src/frontend/src/components/public/PIAIntakeFilter/index.tsx [hunk 2]
```diff
@@ -9,42 +9,45 @@ import {
 } from '../../../constant/constant';
 import { isMPORole } from '../../../utils/helper.util';
 import Dropdown from '../../common/Dropdown';
-import { IFilter } from './interfaces';
 
-const PIAIntakeFilter = ({ id }: IFilter) => {
-  const navigate = useNavigate();
+const PIAIntakeFilter = () => {
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/public/PIAIntakeFilter/index.tsx:13]: Cognitive Complexity of 39 😱

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 3]
```diff
@@ -30,27 +34,65 @@ const PIAList = () => {
...
+    if (searchParams.get('filterPiaDrafterByCurrentUser'))
+      params.filterPiaDrafterByCurrentUser = filterPiaDrafterByCurrentUser;
+    if (searchParams.get('filterByStatus'))
+      params.filterByStatus = filterByStatus;
+    if (searchParams.get('filterByMinistry'))
+      params.filterByMinistry = filterByMinistry;
+
+    for (const [key, value] of searchParams.entries()) {
+      params[key] = value;
+    }
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAListPage/index.tsx:50]: aren't line 41-46, and 48-49 doing same things? 🤔

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 4]
```diff
@@ -7,19 +7,23 @@ import { tableHeadingProperties } from './tableProperties';
...
+  const [searchParams, setSearchParams] = useSearchParams();
   const [SortBy, setSortBy] = useState('');
   const [SortOrder, setSortOrder] = useState(0);
   const [currentPage, setcurrentPage] = useState(1);
   const [headings, setHeading] = useState(tableHeadingProperties);
   const [PageSizedefault, setPageSizedefault] = useState(10);
 
+  const [filterByMinistry] = useState<string>('');
+  const [filterByStatus] = useState<string>('');
+  const [filterPiaDrafterByCurrentUser] = useState<string>('');
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAListPage/index.tsx:26]:
```suggestion
  const [filterByMinistry] = useState<string>(searchParams.get('filterByMinistry') || '');
  const [filterByStatus] = useState<string>(searchParams.get('filterByStatus') || '');
  const [filterPiaDrafterByCurrentUser] = useState<string>(searchParams.get('filterPiaDrafterByCurrentUser') || '');
```

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 5]
```diff
@@ -30,27 +34,65 @@ const PIAList = () => {
...
   };
 
   useEffect(() => {
     const listener = (event: any) => {
       if (event.code === 'Enter' || event.code === 'NumpadEnter') {
         event.preventDefault();
-        navigate(`?searchText=${searchText}`);
+        const params: any = {};
+
+        params.searchText = searchText;
```
Comment (kushal-arora-fw) [inline src/frontend/src/pages/PIAListPage/index.tsx:64]: why not conditional here like for other searchParams?

`if (searchParams.get('searchText')) params.searchText = searchText;`

File: src/frontend/src/pages/PIAListPage/index.tsx (hunk 3 above)
Comment (binggaofw) [reply src/frontend/src/pages/PIAListPage/index.tsx:56]: yes, i am working on this issue. trying to find a way can call function outside of useEffect hook

File: src/frontend/src/pages/PIAListPage/index.tsx [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
-import { faPlus, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
 import EmptyPIAList from '../../components/public/EmptyPIAList';
 import PIAListTable from '../../components/public/PIAListTable';
 import { usePIALookup } from '../../hooks/usePIALookup';
 import { tableHeadingProperties } from './tableProperties';
 import { PiaSorting } from '../../constant/constant';
 import Pagination from '../../components/common/Pagination';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
```
Comment (Bottle7) [inline src/frontend/src/pages/PIAListPage/index.tsx:9]:
```suggestion
import { useCallback, useEffect, useState } from 'react';
```

**Review by Bottle7 — DISMISSED** [review]
Tested. Can use filters on search results ✅

**Review by kushal-arora-fw — APPROVED** [review]

**Review by Bottle7 — APPROVED** [review]

**Pull request merged**